src/metabase/api/search.clj

```diff
@@ -477,26 +431,38 @@
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
 (mu/defn ^:private search-context :- SearchContext
-  [search-string   :- [:maybe ms/NonBlankString]
-   archived-string :- [:maybe ms/BooleanString]
-   table-db-id     :- [:maybe ms/PositiveInt]
-   models          :- [:maybe [:or SearchableModel [:sequential SearchableModel]]]
-   limit           :- [:maybe ms/PositiveInt]
-   offset          :- [:maybe ms/IntGreaterThanOrEqualToZero]]
-  (cond-> {:search-string      search-string
-           :archived?          (Boolean/parseBoolean archived-string)
-           :current-user-perms @api/*current-user-permissions-set*}
-    (some? table-db-id) (assoc :table-db-id table-db-id)
-    (some? models)      (assoc :models
-                               (apply hash-set (if (vector? models) models [models])))
-    (some? limit)       (assoc :limit-int limit)
-    (some? offset)      (assoc :offset-int offset)))
+  [{:keys [archived-string
+           created-by
+           limit
+           models
+           offset
+           search-string
+           table-db-id]} :- [:map {:closed true}
+                             [:search-string                    [:maybe ms/NonBlankString]]
+                             [:archived-string {:optional true} [:maybe ms/BooleanString]]
+                             [:created-by      {:optional true} [:maybe ms/PositiveInt]]
+                             [:limit           {:optional true} [:maybe ms/Int]]
+                             [:models          {:optional true} [:maybe [:or SearchableModel [:sequential SearchableModel]]]]
+                             [:offset          {:optional true} [:maybe ms/Int]]
+                             [:table-db-id     {:optional true} [:maybe ms/PositiveInt]]]]
+  (let [models (if (string? models) [models] models)
+        ctx    (cond-> {:search-string      search-string
+                        :current-user-perms @api/*current-user-permissions-set*
+                        :archived?          (Boolean/parseBoolean archived-string)
+                        :models             (apply hash-set (if (seq models) models search-config/all-models))}
+                 (some? created-by)  (assoc :created-by created-by)
+                 (some? table-db-id) (assoc :table-db-id table-db-id)
+                 (some? limit)       (assoc :limit-int limit)
+                 (some? offset)      (assoc :offset-int offset))]
+    (assoc ctx :models (search.filter/search-context->applicable-models ctx))))
```

Comment:
I think we should still be taking into account `models` from the argument into account?

Reply:
yes it is. if you look into the implementation of `search-context->applicalble-models`, it intersects with `models` from the context at the end.